### DISCUSSION TITLE
[Proposal] 🐄 COWS! 🐄

### DISCUSSION THREAD

**Discussion by Blasta-alasta** [sunflower-land/sunflower-land#2227] [Ideas]
### **COWS!** 🐮🐮🐮

![206883541-bc017bce-1b91-43e6-892c-a7abd9474b23](https://user-images.githubusercontent.com/102267293/218303255-ae7dba6e-55ca-44a8-a525-5d5c5f35fc88.png)

_With the introduction to cows I think it would be good to add some extra features for the players. Below are my ideas which have turned into somewhat of a proposal._

**Game Loop:**
- Build the Cow Shed & Milking Parlour
- Purchase Calf from Cow Shed > Feed > Calf will grow into a type of Cow
- Milk Cows at the Milking Parlour
- Feed Cows Wheat
- Give Treats to Cow increase their Love, increasing Milk Yield
- Follow Cow Life Cycle; Birth, Expiration (Death) and Breeding
- Monitor your Cow’s progress at the Cow Shed
- Use Milk to complete Quests/Race, or to make Milkshakes for your Bumpkins.

> For the purpose of this proposal I have gone with the idea that Cows will require milking every 24 hours, then when complete will require their daily feed of 1 wheat.*
> 
> *Farmers with a Golden Rooster would get this free, as with chickens.

### The Cow Shed ![reg](https://user-images.githubusercontent.com/102267293/218304341-5131252e-6f7c-4c63-8dea-3285dfc56f19.png)

_First in game loop is Building a Cow Shed which acts the same as a Hen House does for chickens, except players will be able to purchase their first Calves.
Players will be able to purchase up to 3 Calves for their farm. A Calf will require a feed of 1 wheat* and then will take 24 hours to grow up into a type of Cow._

**Building Requirements** (suggestion only)

- 300 Wood
- 100 Stone
- 5 Gold
- 15 SFL
- 24 hours to complete
- 4 by 3 square space

**Purchasing a Calf**

- 5 SFL each
- Limit of 3 at a time

### Milking Parlour
_Cows require more buildings than just their barn! If players want milk then a milking parlour must be build!_

**Requirements**
- 50 Wood
- 10 Iron
- 5 SFL
- 3 hours to complete
- 3 by 2 square space

**Milking a Cow**

1. Cows require milking daily, every 24 hours at the Milking Parlour.
2. A Milking Parlour can only hold 1 Cow at a time and takes 5 mins to milk that 1 Cow.
3. After a Cow has been milked he needs his daily feed of 1 wheat to reset the 24 hour timer for milking.

**Milk Yield**

Yield is dependent on a a few factors, which is where a new game mechanism can come in. A Cow that is feed daily will produce 1 Milk Bottle. 
_But a Cow that has been loved and taken care of will produce more…._

### Love & Treats
_Cows require love and attention to be happy producers. I propose a system where players will have to feed their cows extra treats in order to keep them feeling loved and producing their best yield when it comes to milking time!_

**Idea**
Every 6 hours a cow will display a little speech bubble with a picture of a random crop that it wants as a treat. The choice of treat should only be one of the lesser timed crops to grow (from sunflowers, potatoes, pumpkins, carrots, or beetroots), and always just one of each item.

Once a treat is given, the 6 hour timer starts again. Every treat that is given will make the Cow produce another 0.15 Milk Bottles. So a fully satisfied cow will produce 1.6 Milk Bottles at milking time instead of 1.

The window for satisfying a Cow’s craving is only 30 mins, after that the opportunity is missed.

> **Example for one day cycle:**
> 
> - Player logs in at 9am to milk the cows after feeding the day before. Cows are milked at the Milking Parlour and then fed their daily feed of 1 wheat. (Cows were given treats last at 4am)
> 
> - Player logs in at 10am and one of the cows wants a potato, and the other two want a carrot each. Player gives all three their treat. And Cows give hearts! (+0.15 each on next milking)
> 
> - Players logs in at 4:11pm and one want a carrot, one a sunflower, one a pumpkin. Player gives all three their treat. And Cows give hearts! (+0.15 each on next milking)
> 
> - Players logs in at 10:04pm and all three want a pumpkin. Player attempts to gives all three their treat but accidentally misses one because player is tired. And only two Cows give hearts! (+0.15 each those two cows next milking
> 
> - Players doesn’t log in between 4am and 4:30am because he loves his sleep. Cows miss their treat.
> 
> - Player logs in at 9am to milk his cows after feeding the day before. Cows are milked at the Milking Parlour:
> 
> One cow got 2/4 of his daily treat cravings satisfied so it produces his regular 1 bottle of milk plus an extra 0.3
> 
> Two cows got 3/4 of his daily treat cravings satisfied so they each produce their regular 1 bottle of milk plus an extra 0.45 each
> 
> Total Milk Yield for the day 1.3 + 1.45 + 1.45 = 4.2 bottle of milk
> 
> - After Milking is complete, the Cows are fed their daily feed of 1 wheat, and the cycle is repeated.

### Cow Life Cycle
_Chickens might live forever (for now), but sadly Cows don’t. But that’s okay because an important part of a life cycle is not just growing up and eventually expiring…. There’s also Breeding!_

**Life Span**
Once a Calf has grown into a Cow, it will continue to ‘age’ and eventually come to an ‘expiry’… in which case a new Calf will be needed.

**Proposed life span once a Calf becomes a Cow : 30 Days**

This means a Cow (if feed and milked daily) will produce at less 30 Milk Bottles before expiring (up to 48 Milk Bottles, depending on how many treats are feed during it’s lifecycle).

New calves are purchased at the Cow Shed, but they can also be breed.

### Breeding / Bull
_A Bull is an NFT that needs to be placed independently of the Cows. A Bull is purchasable (craft-able) at Helios with an unlimited supply and is non-withdraw-able._

**Requirements For Crafting a Bull**

- 20 Milk Bottles
- 5 Gold

**Idea**
A Bull will keep your Cows breeding. This means once a Cow has come to the end of it's lifetime, a new Calf will be produced and the player will not have to purchase one to replace the one that expires. But a Bull only has enough stamina to produce 2 free Calves every 30 days. (This means the player will still have to purchase 1 new Calf but would get 2 for free when it comes time to replacing the last 3.)

A Bull also does not live forever and will have to replaced (if the the player desires too do so) once it expires. 

A Bulls lifespan is 90 Days, so that during his lifetime he can produce 6 free calves.

When Breeding your Bull with your Cows, a Calf will be produced (on the last day of your Cow’s lifecycle). Once that calf grow up (after 24 hours of being feed) it will have a different percentage chance of being regular Cow or a rare Mutant Cow.

**Note:** Only bred calves have a chance of growing into a Mutant Cow. Calves bought through the Cow Shed can only grow up into regular Cows.

### Mutant Cows
_Mutant Cows will work differently than Mutant Chickens; Mutant Cows will not be a withdrawal NFT on top of your regular animals. Instead a Mutant Cow will take the place of a regular Cow, but give the extra boost associated with that particular Mutant…. but Mutant Cows will also have a limited lifespan - the same amount of days as a regular Cow; 30 days._

**Mutants & Droprates**
A Calf Bred from a Players Bull and Cow will have a chance of growing up to being a Regular Cow _(common - 85% chance)_ or….

- **Scrawny Cow** - _Can live just on treats (no wheat required after milking). (uncommon 1/20 chance)_ 

- **Lovey-Dovey Cow** - _Always produces max yield without any love/treats (uncommon 1/20 chance)_ 
![love](https://user-images.githubusercontent.com/102267293/218304316-61444db7-3dc4-4169-8565-ffa4d4309d9c.png)

- **Chocolate Cow** - _Produces Chocolate Milk (super rare 1/100 chance) (Chocolate Milk Bottles can be drunken straight by the players bumpkin and gives excellence EXP)_
- **Big Moo-Ma Cow** - _Produces double milk (super rare 1/500 chance)_
- **Immortal Cow** - _Never needs feeding, never expires, never needs replacing, the Immortal Cow lives on (ultra rare 1/1000 chance)_

### NFTs
_When Cows are released a similar Race/Quest could be done as was done with the Fruit Release. Some ideas:_

_**Cow Mattress (unlimited - Helios)**_
Well rested cows are happy cows. Cow’s & Mutant Cow’s lifespan increases to 40 Days (+10 Days).
**Requirements**

- 60 Milk Bottles
- 5 Gold

**_Highland Coo (limited 3000 - Goblin Retreat)_**![coo](https://user-images.githubusercontent.com/102267293/218304314-c3b6d746-4daa-4b2f-9154-900468d353ac.png)
The shaggy classic… Reduces cow feeding requirements by 20% _(0.8 wheat, and 0.8 for each treat)_
**Requirements**

- 60 Milk Bottles
- 25 Gold

**_Jack’s Cow (limited to 1000 - Goblin Retreat)_**
Does it need explaining…. Gives discount to Magic Beans (limited 1000) _TBC_
**Requirements**

- 120 Milk Bottles
- 25 Gold

**_Golden Calf Statue (limited 500 - Goblin Retreat)_**
Cows admire this magical statue, found by the Bumpkins of old…. 
Increase Cow Shed capacity from 3 to 5
Milking Time at the Parlour is reduced by 50%
Increase bulls stamina from 2 to 4 calves per 30 days
**Requirements**

- 300 Milk Bottles
- 50 Gold
- 50 SFL

**_Golden Handle Milk Bucket (Bumpkin handheld item - limited to 1000 - Goblin Retreat)_**
Boost: Reduce treat cravings from Cows to 3 a day / every 8 hours / +0.2 milk per treat
**Requirements**

- 10 Orange Shakes
- 10 Red Shakes
- 10 Super Shakes
- 10 Iron
- 10 Gold

### Quest Bumpkin Items:
_Along with the above NFTs, related Bumpkin Items could also be introduced alongside Cows, again as was done with the Fruit Race/Quest. Some ideas:_

**Farmers Gumboots (Bumpkin boots)**

- Milk Cows 10 times

**Lucky Cow Charm (Bumpkin necklace)** 
Boost: Doubles the chances off mutant cows

- Make 20 Milkshakes

**Bull Mask/Shirt (Bumpkin Mask/Shirt)**

- Breed 20 Calves

### Uses for Milk (coming soon)

**Bumpkin Milkshakes**

_Sun Shake		3 Milk Bottle (MB), 500 Sunflower
Snip Shake		2 MB, 10 Parsnip
Blue Shake 		2 MB, 10 Blueberry
Orange Shake 	2 MB, 4 Orange, 5 Pumpkin, 5 Carrot
Red Shake  		2 MB, 2 Apple, 10 Radish
Super Shake		4 MB 5 Kale_

**More Foods such as Cheese (Deli), Chowder (Kitchen) etc.**

**Withdrawable and tradable** _(after Milk NFT Race/Quests are done)_

![Regbell](https://user-images.githubusercontent.com/102267293/218304360-081b67ae-392e-4e5c-a17a-a32c8b631ddb.png) _These are just ideas. Expect any values or mechanisms to change if anything from this introduced. Also forgive the art... i'm a real amatuer._ ![Regbell](https://user-images.githubusercontent.com/102267293/218304360-081b67ae-392e-4e5c-a17a-a32c8b631ddb.png)

**Comment by Alaindrea** [comment]
since bulls are male cows, every breeding should have a chance to produce a bull. excess bulls can make steak/ground beef. You should rent a bull instead of buying one in the beginning, to prevent players buying them for steak. It would be better if bulls and cows didn't expire, but maybe the cows only produce milk for a certain amount of time, before they need to spend time with a bull again.  Keeping bulls on your farm should require a pasture. It is crazy to only feed cows 1 wheat.  Cows should need 2-3 wheat.
p.s. I really like you cow art

**Reply by Blasta-alasta** [reply]
Awesome... Thanks for all the feedback! So happy you liked the cows too lol. But also.... aren't Bumpkins Vegetarians?

**Reply by Alaindrea** [reply]
says who?

**Comment by IvanSkazkin** [comment]
I see the value of a golden rooster being able to feed any animal that requires wheat. It's logical

**Reply by Blasta-alasta** [reply]
I see value too, in saying that I check the description and it says "feed chickens without wheat" so probably shouldn't apply to other animals. Anyway Spencer has made an official proposal for Cows now here https://github.com/sunflower-land/sunflower-land/discussions/2240

**Comment by HootWise** [comment]
How about feeding just once every 72 hours instead of once a day?
